testsuite/MDAnalysisTests/analysis/test_base.py

```diff
@@ -182,15 +80,104 @@ def u():
+    (SerialOnly, 'dask', None),
+    (ParallelizableWithDaskOnly, 'multiprocessing', None),
+    (ParallelizableWithDaskOnly, 'multiprocessing', 2),
+])
+def test_backend_configuration_fails(u, run_class, backend, n_workers):
+    u = mda.Universe(TPR, XTC)  # dt = 100
+    with pytest.raises(ValueError):
+        _ = run_class(u.trajectory).run(backend=backend, n_workers=n_workers, stop=0)
+
+@pytest.mark.parametrize('run_class,backend,n_workers', [
+    (Parallelizable, CustomSerialBackend, 2),
+    (ParallelizableWithDaskOnly, CustomSerialBackend, 2),
+])
+def test_backend_configuration_works_when_unsupported_backend(u, run_class, backend, n_workers):
+    u = mda.Universe(TPR, XTC)  # dt = 100
+    backend_instance = backend(n_workers=n_workers)
+    _ = run_class(u.trajectory).run(backend=backend_instance, n_workers=n_workers, stop=0, unsupported_backend=True)
+
+@pytest.mark.parametrize('run_class,backend,n_workers', [
+    (Parallelizable, CustomSerialBackend, 1),
+    (ParallelizableWithDaskOnly, CustomSerialBackend, 1),
+])
+def test_custom_backend_works(u, run_class, backend, n_workers):
+    backend_instance = backend(n_workers=n_workers)
+    u = mda.Universe(TPR, XTC)  # dt = 100
+    _ = run_class(u.trajectory).run(backend=backend_instance, n_workers=n_workers, unsupported_backend=True)
+
+@pytest.mark.parametrize('run_class,backend_instance,n_workers', [
+    (Parallelizable, map, 1),
+    (SerialOnly, list, 1),
+    (ParallelizableWithDaskOnly, object, 1),
+])
+def test_fails_incorrect_custom_backend(u, run_class, backend_instance, n_workers):
+    u = mda.Universe(TPR, XTC)  # dt = 100
+    with pytest.raises(ValueError):
+        _ = run_class(u.trajectory).run(backend=backend_instance, n_workers=n_workers, unsupported_backend=True)
+
+    with pytest.raises(ValueError):
+        _ = run_class(u.trajectory).run(backend=backend_instance, n_workers=n_workers)
+
+@pytest.mark.parametrize('run_class,backend,n_workers', [
+    (SerialOnly, CustomSerialBackend, 1),
+    (SerialOnly, 'multiprocessing', 1),
+    (SerialOnly, 'dask', 1),
+])
+def test_fails_for_unparallelizable(u, run_class, backend, n_workers):
+    u = mda.Universe(TPR, XTC)  # dt = 100
+    with pytest.raises(ValueError):
+        if not isinstance(backend, str):
+            backend_instance = backend(n_workers=n_workers)
+            _ = run_class(u.trajectory).run(backend=backend_instance, n_workers=n_workers, unsupported_backend=True)
+        else:
+            _ = run_class(u.trajectory).run(backend=backend, n_workers=n_workers, unsupported_backend=True)
 
 @pytest.mark.parametrize('run_kwargs,frames', [
     ({}, np.arange(98)),
     ({'start': 20}, np.arange(20, 98)),
     ({'stop': 30}, np.arange(30)),
     ({'step': 10}, np.arange(0, 98, 10))
 ])
-def test_start_stop_step(u, run_kwargs, frames):
-    an = FrameAnalysis(u.trajectory).run(**run_kwargs)
+def test_start_stop_step(u, run_kwargs, frames, client_FrameAnalysis):
```

Comment:
This seems like a pretty crucial test that it might be worth doing two copies of this? One where you do pass the new fixture and one where you don't? The thought here is that we want to ensure that we don't introduce breaking behaviour anywhere by having some kind of non optional new kwarg.

Reply:
In this particular case -- sure, no problem. But duplicating all tests for every backend seems very excessive, and I'd rather not do that.

Reply:
Done -- and I also done it for `test_frame_slice`

Reply:
Added